spec.html

```diff
@@ -43062,7 +43064,14 @@ <h1>
             1. Return ! CreateIterResultObject(_value_, _done_).
           1. Let _onFulfilled_ be ! CreateBuiltinFunction(_unwrap_, 1, *""*, &laquo; &raquo;).
           1. NOTE: _onFulfilled_ is used when processing the *"value"* property of an IteratorResult object in order to wait for its value if it is a promise and re-package the result in a new "unwrapped" IteratorResult object.
-          1. Perform ! PerformPromiseThen(_valueWrapper_, _onFulfilled_, *undefined*, _promiseCapability_).
+          1. If _closeIteratorOnRejection_ is *false*, then
```

Comment:
Probably this should also have `, or if _done_ is *true*`, to handle the case that the rejected promise is the last value produced by the iterator (as would happen if it did `return Promise.reject()` rather than `yield Promise.reject()`).

Reply:
Wouldn't that case be handled by the fact that `done === true` would be the result of a `next` call, so `closeIteratorOnRejection` would already be true?

Reply:
Like I don't think we should proactively close on done, only if it's a rejection. If it's a normal result, we should let the consumer of the async iterator initiate the close (potentially with a value).

Reply:
I think you've misread my comment. I'm saying we should change this `if` from `If _closeIteratorOnRejection_ is *false*, then let _onRejected_ be *undefined*.` to `If _closeIteratorOnRejection_ is *false*, or if *done* is true, then let _onRejected_ be *undefined*.`

My point is that if the sync iterator does `return Promise.reject()`, then we should _not_ close the sync iterator once the promise rejects, because it's already finished.

The logic currently would call IteratorClose in that case, and my suggestion would stop that from happening.

Reply:
Oh yes sorry my bad, totally mixed it up, long day!